### PR TITLE
Add CONSENT a self long read corrector v2.1

### PR DESCRIPTION
**Pull request by natir**
Add CONSENT a self corrector long-reads.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by natir** [comment]
@BiocondaBot please update

**Comment by natir** [comment]
@BiocondaBot please update

**Comment by natir** [comment]
Hello everyone,

Ok for this package I made many bad hack, we can probably found a better way, to solve all issues.

Create a package for dependencies could be a good solution but I'm pretty sure dependencies of this package isn't useful for other people and I prefer not to use my time to create these packages.

My main question is why the build failed on docker image build?

**Comment by natir** [comment]
@BiocondaBot please update

**Comment by natir** [comment]
@BiocondaBot please fetch artifacts

**Comment by natir** [comment]
@BiocondaBot please add label

**Review by innovate-invent — CHANGES_REQUESTED** [review]

File: recipes/consent/build.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+cd CTPL
+git checkout 437e135dbd94eb65b45533d9ce8ee28b5bd3
+cd ..
+
+# Fix some error can't be patch before
+if [ "$(uname)" == "Darwin" ]; then
+    sed -i '' -e "s#/BMEAN/BOA/blosum80.mat#/../share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/blosum80.mat#" src/main.cpp # Fix matrix path
+else
+    sed -i'' -e "s#/BMEAN/BOA/blosum80.mat#/../share/$PKG_NAME-$PKG_VERSION-$PKG_BUILDNUM/blosum80.mat#" src/main.cpp # Fix matrix path
+fi
```
[the remark this answers is not in the document]
Comment (natir) [reply recipes/consent/build.sh:25]: env variable $PKG_NAME, $PKG_VERSION, $PKG_BUILDNUM can be interpeted durring patch step ?

File: recipes/consent/build.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+cd spoa
+mkdir build
+cd build
+cmake -DCMAKE_BUILD_TYPE=Release ..
+make
+cd ../..
+
+make CXX=$CXX # BMEAN make
+cd ..
```
[the remark this answers is not in the document]
Comment (natir) [reply recipes/consent/build.sh]: I try to simplify the cmake part but we can maybe found a better way

File: recipes/consent/build.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+git clone --recursive https://github.com/Malfoy/BMEAN.git
+cd BMEAN
+git checkout 40ab186c106b2c8303a5605e21d3f4a2e15f809e
+cd ..
+
+rm -rf CTPL
+git clone --recursive https://github.com/vit-vit/CTPL.git
+cd CTPL
+git checkout 437e135dbd94eb65b45533d9ce8ee28b5bd3
+cd ..
```
[the remark this answers is not in the document]
Comment (natir) [reply recipes/consent/build.sh]: done

File: recipes/consent/build.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (innovate-invent) [reply recipes/consent/build.sh:19]: I see, that is unfortunate.

File: recipes/consent/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (innovate-invent) [reply recipes/consent/build.sh]: I am not sure about cmake. Perhaps the following:
```
cmake -DCMAKE_BUILD_TYPE=Release BMEAN/spoa
make -C BMEAN/spoa/build
```

[the remark this answers is not in the document]
Comment (natir) [reply recipes/consent/build.sh]: It needs another option but I found it

**Review by innovate-invent — APPROVED** [review]
Looks great, thanks for the contribution!

**Comment by natir** [comment]
@BiocondaBot please merge

**Pull request merged**